ESCPOS_NET/Utils/DeviceFinder.cs

```diff
@@ -0,0 +1,211 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace ESCPOS_NET
+{
+    public class DeviceFinder
+    {
+        #region Win API
+        private struct SP_DEVICE_INTERFACE_DATA
+        {
+            internal int cbSize;
+
+            internal Guid InterfaceClassGuid;
+
+            internal int Flags;
+
+            internal IntPtr Reserved;
+        }
+        private struct SP_DEVINFO_DATA
+        {
+            internal int cbSize;
+
+            internal Guid ClassGuid;
+
+            internal int DevInst;
+
+            internal IntPtr Reserved;
+        }
+        [StructLayout(LayoutKind.Sequential, Pack = 1)]
+        private struct DEVPROPKEY
+        {
+            public Guid fmtid;
+            public uint pid;
+        }
+        private static readonly IntPtr INVALID_HANDLE_VALUE = new IntPtr(-1);
+        [DllImport("setupapi.dll", CharSet = CharSet.Unicode, SetLastError = true)]
```

Comment:
Hey all, sorry I missed this.  How does this work outside of Windows? This seems like old school COM stuff that's windows / full framework only (not .NET core / .NET standard)

Reply:
This is like USB1, USB2.... Planning to mergue it ???

Reply:
@lukevp I don't think those imports will work outside windows. From experience with Mono back in the day one would need to have corresponding .so files and etc. Nothing straight forward.

Reply:
But, it could be a Windows-only feature, since it seems that only Windows has this issue with USB. Thoughts?

Reply:
Hi all, 
I developed this feature as a windows only feature, because there wasn't any way of printing to USB printer on Windows without using virtual serial port drivers.
@lukevp As far as I understand Devicefinder.cs doesn't have any other dependency outside .netStandard but yes it only works on windows (finding devices connected).